src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs

```diff
@@ -206,6 +206,14 @@ public async Task<ActionResult<List<ReleaseViewModel>>> ListDraftReleases()
                 .HandleFailuresOrOk();
         }
 
+        [HttpGet("releases/approvals")]
+        public async Task<ActionResult<List<ReleaseViewModel>>> ListUsersReleasesForApproval()
```

Comment:
I've got a few issues with using `ReleaseViewModel` as it currently stands:

- It has properties around things like prerelease that we don't need
- There are properties like `updatePublishedDate` and `notifySubscribers` that are literally only relevant on the release status form
- Publication properties are exposed like `publicationSlug`, `publicationTitle`, etc, rather than be aggregated through a `publication` property.

It feels like this view model is basically a bit of a trainwreck currently and needs some serious cleanup.

However, we don't need to do this as part of this PR as we should be able to just use `ReleaseSummaryViewModel` instead, with a slight modification around the publication:

```cs
// Probably could be a record...
public class ReleaseSummaryViewModel
{
    public PublicationSummaryViewModel? Publication { get; set; }
}
```

The `publication` property would be intentionally nullable so we can choose to include or not on a case-by-case basis. In this case, we would need it to be included.

Would suggest we do this with the other endpoints for draft and scheduled releases as well.

This will need an appropriate change in the frontend too.

Reply:
I'm happy to shift to using something other than ReleaseViewModel, but I'd suggest perhaps leaving it out of this particular PR and addressing it in a separate piece of work, for 2 reasons.  One is that it expands the scope of the testing of this ticket and adds an additional piece of risk, and another is that I've started to look into this change and realise that a wider view of the problem might be worth investing time in.

When I started to look into switching over to ReleaseSummaryViewModel, I saw that we're already doing conditional includes of additional information in it (e.g. Permissions), and it feels like we're perhaps in danger of muddying the waters more by adding more conditional information to it.  Meanwhile, ReleaseViewModel continues to be its own separate thing entirely but with a lot of overlap.

What I'd suggest is that we create a new ticket to investigate how we could rationalise all of this - perhaps we don't need a ReleaseSummaryViewModel and a ReleaseViewModel anymore, each with their subtle differences.  Maybe we have 1 view model with core Release information on, and a bunch of conditional information that can be controlled via request parameters, and we do it in one consistent way.  There would be less top-level individual fields in this view model, and instead I'd expect them to be nested properties depending on what you've asked for e.g. if we ask for the release view model plus pre-release information, `PreReleaseAccessList` and `PreReleaseUsersOrInvitesAdded` fields will be included under an overarching "PreRelease" parent.  We could then look at finding if there are better ways of achieving this conditional mapping with AutoMapper than we are already doing (whereby afaik we're currently letting AutoMapper map the core stuff and then we're manually adding extra information afterwards - I suspect there's probably a neater way to do it).

Are you happy that we take this approach with a new investigation ticket?

Reply:
Personally, I'd pretty strongly prefer to swap over to `ReleaseSummaryViewModel` now, at least as a temporary measure. As the dashboard is a hot path, it's not great that we're returning unnecessarily bloated view models and have been doing for some time. For context, on dev, the draft releases tab returns over 400 releases.

I think a more heavy handed approach with an investigation ticket is essentially kicking this into the long grass. It feels unlikely we'd prioritise this, so I think it'd be better to get some benefit now than come back to this much later down the line.

I've looked into this change and it's really not a big one in terms of impact. I've actually knocked together most of the changes and can commit this if you'd prefer to not have to deal with it yourself?

Reply:
Haha no it's OK I don't mind - I've still got my partial work from before kicking around too so I'll resurrect that, but thanks for the offer and for the effort.

Reply:
Okay no problem. Here's the commit anyway if you want to cherry pick it: https://github.com/dfe-analytical-services/explore-education-statistics/commit/962d77fa80fc7164c929d2f6404d191a2644485c

Note that I didn't finish it all off so there's a few broken bits of test data in the frontend still.

Reply:
Nice one, appreciate it :)